spotify_to_ytmusic/spotify.py

```diff
@@ -91,6 +89,16 @@ def getLikedPlaylist(self):
             "description": "Your liked tracks from spotify",
         }
 
+    __id_extraction_regex = re.compile(r"playlist\/(?P<id>\w{22})\W?")
+
+    @classmethod
+    def getPlaylistIdFromUrl(cls, url) -> str:
+        if (match := cls.__id_extraction_regex.search(url)):
+            return match.group("id")
+        elif (match := re.search(r"playlist\/(?P<id>\w+)\W?", url)):
```

Comment:
Why do you think a custom regex can do a better job than Python's battle-tested urllib?

Please show a case that can't be handled with an urllib implementation.

Reply:
I do realize that the existing implementation was not ideal, but I'd like to keep using urllib.urlparse unless you have a significant reason for not doing so

Reply:
Tbh because the existing version wasn't working and looking at it, it was surprising to me that it wasn't working -- why would a query parameter influence what is inside `path`.
I'm definitely with you on not reinventing the wheel with URL parsing, but this regex doesn't parse the URL, it just extracts the ID part - it doesn't make more assumptions than the existing code (existing version requires valid URL, `playlist/id{22}`; this version just requires `playlist/id{22}`)

Reply:
I could run the regex on `url_paths.path` instead of the URL directly, but I think that's more complex than just not parsing the URL for the same result.

Reply:
Would you accept this?

```python
    from urllib.parse import urlparse

    ...

    __id_extraction_regex = re.compile(r"playlist\/(?P<id>\w{22})")

    @classmethod
    def getPlaylistIdFromUrl(cls, url) -> str:
        parsed = urlparse(url)
        if (match := cls.__id_extraction_regex.search(parsed.path)):
            return match.group("id")
        elif (match := re.search(r"playlist\/(?P<id>\w+)", parsed.path)):
            raise ValueError(f"Bad playlist id: {id}")
        else:
            raise ValueError(f"No id found in playlist url: {url}")
```

Reply:
OK I guess we can go with the regex if you think it's simpler. 

Few more comments:

- why is the regex defined on class level, just put it inside the function. The speed improvement gained from compilation is definitely negligible in this case, if there even is any improvement
- why is there a need for the "Bad playlist id" fallback? Is it just for informing the user that the URL is ok but maybe missed a few characters?

Reply:
> why is there a need for the "Bad playlist id" fallback? Is it just for informing the user that the URL is ok but maybe missed a few characters?
yeah, it existed before, I just moved it into this method